migrate.py

```diff
@@ -705,13 +705,12 @@ def read_module_txt_n_fst(path):
 
 
 def inject_init_into_tree(target_dir):
-    for initpath in (
-            os.path.join(dp, '__init__.py')
-            for dp, dn, fn in os.walk(target_dir)
-            if '__init__.py' not in fn
-            # and any(f.endwith('.py') for f in fn)
-    ):
-        write_text_into_file(initpath, '')
+    """Insert ``__init__.py`` file into unit test directories."""
+    for root, dirs, files in os.walk(target_dir):
+        python_files_exist = any(fn.endswith('.py') for fn in files)
+        dirs_exist = len(dirs) > 0
+        if '__init__.py' not in files and (python_files_exist or dirs_exist):
```

Comment:
Let's add an explanation for this condition too, for consistency/readability:
```suggestion
        is_package_dir = '__init__.py' in files
        if not is_package_dir and (python_files_exist or dirs_exist):
```

Reply:
`is_package_dir` does not really explain anything to me. I think it's clear as is.

Reply:
well, checking that init is in some list of files doesn't communicate the intent either

Reply:
If you look only at that specific line than probably not. But the function is called 'inject_init_into_tree`. Also after checking for missing `__init__.py` we create it. I am not sure how is that not clear in the context of that function.